kitsune/sumo/static/sumo/js/tests/form-wizard-sign-in-step-tests.js

```diff
@@ -0,0 +1,305 @@
+      utm_source: { type: "hidden", value: "utm_source", disabled: false },
+      utm_campaign: { type: "hidden", value: "utm_campaign", disabled: false },
+      utm_medium: { type: "hidden", value: "utm_medium", disabled: false },
+      entrypoint: { type: "hidden", value: "entrypoint", disabled: false },
+      entrypoint_experiment: { type: "hidden", value: "entrypoint_experiment", disabled: false },
+      entrypoint_variation: { type: "hidden", value: "entrypoint_variation", disabled: false },
+      context: { type: "hidden", value: "context", disabled: false },
+      redirect_to: { type: "hidden", value: window.location.href, disabled: false },
+      redirect_immediately: { type: "hidden", value: "true", disabled: false },
+
+      service: { type: "hidden", value: "sync", disabled: false },
+      action: { type: "hidden", value: "email", disabled: false },
+      flow_id: { type: "hidden", value: "", disabled: true },
+      flow_begin_time: { type: "hidden", value: "", disabled: true },
+      email: { type: "email", value: "", disabled: false },
+    };
+
+    step.setState(TEST_STATE_NO_FLOW_METRICS);
+
+    let form = step.shadowRoot.querySelector("form");
+    assertFormElements(form, EXPECTED_FORM_ELEMENTS_NO_FLOW_METRICS);
+
+    // Now add the flow metrics, which should enable the fields.
+
+    const TEST_STATE_WITH_FLOW_METRICS = {
+      utm_source: "utm_source",
+      utm_campaign: "utm_campaign",
+      utm_medium: "utm_medium",
+      entrypoint: "entrypoint",
+      entrypoint_experiment: "entrypoint_experiment",
+      entrypoint_variation: "entrypoint_variation",
+      context: "context",
+      redirect_to: window.location.href,
+      redirect_immediately: true,
+
+      flow_id: "flow_id",
+      flow_begin_time: "flow_begin_time",
+    };
+
+    const EXPECTED_FORM_ELEMENTS_WITH_FLOW_METRICS = {
+      utm_source: { type: "hidden", value: "utm_source", disabled: false },
+      utm_campaign: { type: "hidden", value: "utm_campaign", disabled: false },
+      utm_medium: { type: "hidden", value: "utm_medium", disabled: false },
+      entrypoint: { type: "hidden", value: "entrypoint", disabled: false },
+      entrypoint_experiment: { type: "hidden", value: "entrypoint_experiment", disabled: false },
+      entrypoint_variation: { type: "hidden", value: "entrypoint_variation", disabled: false },
+      context: { type: "hidden", value: "context", disabled: false },
+      redirect_to: { type: "hidden", value: window.location.href, disabled: false },
+      redirect_immediately: { type: "hidden", value: "true", disabled: false },
+
+      service: { type: "hidden", value: "sync", disabled: false },
+      action: { type: "hidden", value: "email", disabled: false },
+      flow_id: { type: "hidden", value: "flow_id", disabled: false },
+      flow_begin_time: { type: "hidden", value: "flow_begin_time", disabled: false },
+      email: { type: "email", value: "", disabled: false },
```

Comment:
```suggestion
      ...EXPECTED_FORM_ELEMENTS_NO_FLOW_METRICS,
      flow_id: { type: "hidden", value: "flow_id", disabled: false },
      flow_begin_time: { type: "hidden", value: "flow_begin_time", disabled: false },
```

Again optional, but we can be consistent. I do kind of like how this makes it clearer what changed.

Reply:
Oh yes, this is much better, thank you. :)